### PR TITLE
feat(UI, port): Port hiding traits/mutations from DDA

### PR DESCRIPTION
**Pull request by RobbieNeko**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

### Optional

- [x] This PR ports commits from DDA or other cataclysm forks.
  - [x] I have attributed original authors in the commit messages adding [`Co-Authored-By`](https://docs.github.com/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) in the commit message.
  - [x] I have linked the URL of original PR(s) in the description.
- [x] This is a C++ PR that modifies JSON loading or behavior.

## Purpose of change

Much like with hiding clothing, hiding traits and mutations is a much-desired feature for various reasons. Considering how much it's desired and how simple it was to port, I see no reason not to bring it in.

## Describe the solution

Ports https://github.com/CleverRaven/Cataclysm-DDA/pull/52058 from DDA.

## Describe alternatives you've considered

- Reinvent the wheel
- Implode
- Explode
- Wait for someone else to do it

## Testing

It builds, it works (on my machine anyway)
image
image

## Additional context

2/3 of the hiding visuals ports that RoyalFox threw at us done. Next up, Bionics! (For whoever ends up taking that one on)

Don't worry Chaos, this time I made sure the keybind was BN-ified the first time ;P

### CODE REVIEW

**Review by RoyalFox2140 — APPROVED** [review]
Needs a windows user for testing but this should be fine, it builds off the previous PR.

File: src/mutation_ui.cpp [hunk 1]
```diff
@@ -39,6 +39,7 @@ enum class mutation_menu_mode {
     activating,
     examining,
     reassigning,
+    hidding,
```
Comment (chaosvolt) [inline src/mutation_ui.cpp:42]:
```suggestion
    hiding,
```

File: src/mutation_ui.cpp [hunk 2]
```diff
@@ -78,9 +79,14 @@ static void show_mutations_titlebar( const catacurses::window &window,
                           c_light_blue ) + "  " + shortcut_desc( _( "%s to activate mutation, " ),
                                   ctxt.get_desc( "TOGGLE_EXAMINE" ) );
     }
+    if( menu_mode == mutation_menu_mode::hidding ) {
+        desc += colorize( _( "Hidding" ), c_cyan ) + "  " + shortcut_desc( _( "%s to activate mutation, " ),
```
Comment (chaosvolt) [inline src/mutation_ui.cpp:83]:
```suggestion
    if( menu_mode == mutation_menu_mode::hiding ) {
        desc += colorize( _( "Hiding" ), c_cyan ) + "  " + shortcut_desc( _( "%s to activate mutation, " ),
```

File: src/mutation_ui.cpp [hunk 3]
```diff
@@ -440,6 +451,9 @@ detail::mutations_ui_result detail::show_mutations_ui_internal( Character &who )
                         // Describing mutations, not activating them!
                         examine_id = mut_id;
                         break;
+                    case mutation_menu_mode::hidding:
```
Comment (chaosvolt) [inline src/mutation_ui.cpp:454]:
```suggestion
                    case mutation_menu_mode::hiding:
```

File: src/mutation_ui.cpp [hunk 4]
```diff
@@ -602,6 +616,9 @@ detail::mutations_ui_result detail::show_mutations_ui_internal( Character &who )
                             // Describing mutations, not activating them!
                             examine_id = mut_id;
                             break;
+                        case mutation_menu_mode::hidding:
```
Comment (chaosvolt) [inline src/mutation_ui.cpp:619]:
```suggestion
                        case mutation_menu_mode::hiding:
```

File: src/mutation_ui.cpp [hunk 5]
```diff
@@ -612,6 +629,9 @@ detail::mutations_ui_result detail::show_mutations_ui_internal( Character &who )
                 menu_mode = menu_mode == mutation_menu_mode::activating ?
                             mutation_menu_mode::examining : mutation_menu_mode::activating;
                 examine_id = std::nullopt;
+            } else if( action == "TOGGLE_SPRITE" ) {
+                menu_mode = mutation_menu_mode::hidding;
```
Comment (chaosvolt) [inline src/mutation_ui.cpp:633]:
```suggestion
                menu_mode = mutation_menu_mode::hiding;
```

**Review by chaosvolt — COMMENTED** [review]
Compiled and tested, it works but has a spelling error.
image

**Review by chaosvolt — APPROVED** [review]

**Pull request merged**